.github/workflows/build-sdk.yml

```diff
@@ -0,0 +1,106 @@
+###################################################################################################
+###                  THIS IS A REUSABLE WORKFLOW TO BUILD THE SCALA LAUNCHERS                   ###
+### HOW TO USE:                                                                                 ###
+###   - THSI WORKFLOW WILL PACKAGE THE ALL THE LAUNCHERS AND UPLOAD THEM TO GITHUB ARTIFACTS    ###
+###                                                                                             ###
+### NOTE:                                                                                       ###
+###   - SEE THE WORFLOW FOR THE NAMES OF THE ARTIFACTS                                          ###
+###################################################################################################
+
+
+name: Build Scala Launchers
+run-name: Build Scala Launchers
+
+on:
+  workflow_call:
+    inputs:
+      java-version:
+        type       : string
+        required   : true
+    outputs:
+      universal-id:
+        description: ID of the `universal` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.universal-id }}
+      linux-x86_64-id:
+        description: ID of the `linux x86-64` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.linux-x86_64-id }}
+      linux-aarch64-id:
+        description: ID of the `linux aarch64` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.linux-aarch64-id }}
+      mac-x86_64-id:
+        description: ID of the `mac x86-64` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.mac-x86_64-id }}
+      mac-aarch64-id:
+        description: ID of the `mac aarch64` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.mac-aarch64-id }}
+      win-x86_64-id:
+        description: ID of the `win x86-64` package from GitHub Artifacts (Authentication Required)
+        value      : ${{ jobs.build.outputs.win-x86_64-id }}
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      universal-id    : ${{ steps.universal.outputs.artifact-id }}
+      linux-x86_64-id : ${{ steps.linux-x86_64.outputs.artifact-id }}
+      linux-aarch64-id: ${{ steps.linux-aarch64.outputs.artifact-id }}
+      mac-x86_64-id   : ${{ steps.mac-x86_64.outputs.artifact-id }}
+      mac-aarch64-id  : ${{ steps.mac-aarch64.outputs.artifact-id }}
+      win-x86_64-id   : ${{ steps.win-x86_64.outputs.artifact-id }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
```

Comment:
were you going to switch to `setup-sbt`?

Reply:
We should when github the image we use drops sbt, I didn't want to attempt it because of possible side effects